draft-ietf-gnap-core-protocol.md

```diff
@@ -3696,12 +3707,22 @@ fHI6kqm3NCyCCTihe2ck5RmCc5l2KBO/vAHF0ihhFOOOby1v6qbPHQcxAU6rEb907
 
 ~~~
 
+The verifier compares the TLS client certificate presented during 
+mutual TLS negotiation to the expected key of the signer. Since the
+TLS connection covers the entire message, there are no additional
+requirements to check.
+
+Note that in many instances, the verifier will not do a full certificate
```

Comment:
`some instances` would be more appropriate?

Reply:
I think "many" is the right level here, to be honest -- since the protocol has lots of support for key-introduction. Either way it's just an implementation note, and we'll need security considerations on trusted/untrusted certificates and keys.

Reply:
I don't like the way this paragraph is worded. We are giving implementors a license to use self-signed client certs, without giving them guidance on when this is appropriate and when it is not. IMO by default the verifier MUST do full cert chain validation, and there are some cases when it MAY skip it.

Reply:
I disagree with that precedence and with the normative statements -- since GNAP allows for key introduction both at runtime and during a pre-registration, the chain validation only fits limited deployment use cases where the AS has a CA for managing client certs. Also we're talking about validating the _client_ cert here, not the server cert. We can probably benefit from the language in RFC8705 (OAuth MTLS) which has similar considerations. 

It might make more sense to have this discussion in the security considerations as opposed to an aside here, though.

Reply:
To the best of my knowledge, MTLS only (mostly?) makes sense if you have a CA somewhere, otherwise it is not very scalable. And if any normative language is required, the Security Considerations is not the right place for it.

Reply:
Traditional MTLS only scales with a CA because of its dependence on PKI. The trust root is different with this protocol because it works at a different layer.

Just about all of the deployments of OAuth MTLS I've personally been involved in haven't used PKI or CA's. Instead, client apps create and register a cert. This cert gets compared to the registered cert on the connection at the application layer, not at the TLS layer. On the TLS layer it's just assuring that the certificates in use in the connections are the ones indicated. It's a separate question whether or not those certificates are trusted or represent a specific known piece of software, and that question isn't answered at the TLS layer.

It's really analogous to having a non-PKI key and using a different signing mechanism. We want to have both options here for MTLS, and it makes sense to have guidance on the different modalities.

Reply:
OK, makes sense. I was thinking of service mesh deployments (e.g. Istio) where you do have a small local CA.

Please create an issue for moving this discussion to Sec Cons then.

Reply:
A mini-CA is definitely something the AS could use to manage and limit the keys/certs it sees from clients, especially since cert-based keys could be used for the HTTP Signing or JOSE methods, too. 

New issue created to track this at #253